src/Native/System.Native/pal_networking.cpp

```diff
@@ -27,6 +27,11 @@
 #include <unistd.h>
 #include <vector>
 
+#if !HAVE_IN_PKTINFO
+typedef in_addr in_pktinfo;
+#define IP_PKTINFO IP_RECVDSTADDR
+#endif
+
```

Comment:
It's really confusing me below that in_pktinfo has to be used completely differently than when typedef'ed here.